app/controllers/api/v0/products_controller.rb

```diff
@@ -13,7 +13,7 @@ class ProductsController < Api::V0::BaseController
       skip_authorization_check only: [:show, :bulk_products, :overridable]
 
       def show
-        @product = find_product(params[:id])
+        @product = Spree::Product.find_by_id(current_api_user, params)
```

Comment:
This is confusing because Rails also defines `find_by_id` and other `find_by_` methods for attributes. Also, this is implemented as scope but returning a single value. Scopes should return ActiveRecord relations though. I don't actually see why you go through the model. Can't you call the query object directly?

Reply:
Yes, scopes belong to the model. But there can be exceptions if it's a bit more complicated and only relevant for a certain domain. And the finders are not scopes.

Reply:
I renamed the functions and removed the finders from the model.